### PR TITLE
rst-ify README.llvm

### PR DESCRIPTION
**Pull request by mppf**
README.llvm is still pretty accurate.

I reworked the first section in order to reduce confusion about the difference
between llvm support in the compiler and extern blocks and the
experimental --llvm code generation.

### CODE REVIEW

File: doc/release/technotes/README.llvm.rst [hunk 1]
```diff
@@ -33,6 +37,7 @@ generation, and also support --llvm-wide-opt:
   make # you might want to do e.g. make -j 16 for a parallel build
 
 Note:
+
 * If you have a built llvm in third-party/llvm/install, even if you
   forget to export CHPL_LLVM=llvm, the default will be to use the built llvm.
```
Comment (ben-albrecht) [inline doc/release/technotes/README.llvm.rst]: Do you think we should add some \`\`s around the paths and variable names? i.e. `export CHPL_LLVM=llvm` & `third-party/llvm/install`

Comment (mppf) [inline doc/release/technotes/README.llvm.rst]: Sure, will do.

File: doc/release/technotes/README.llvm.rst [hunk 2]
```diff
@@ -51,63 +56,65 @@ Note:
...
+
 * chpl__module.bc is the version that will be linked
 * chpl__module-nopt.bc is the generated code without optimizations applied.
 
-Passing --fast will cause LLVM optimizations to run.
+Passing ``--fast`` will cause LLVM optimizations to run.
 
-The --ccflags option can control which LLVM optimizations are run, using the
+The ``--ccflags`` option can control which LLVM optimizations are run, using the
 same syntax as flags to clang.
```
Comment (ben-albrecht) [inline doc/release/technotes/README.llvm.rst]: couple of more \`\`s

File: doc/release/technotes/README.llvm.rst [hunk 3]
```diff
@@ -51,63 +56,65 @@ Note:
...
 wide pointer, but CHPL_WIDE_POINTERS=node16 packs wide pointers into a
 pointer-sized value (normally 64 bits) which includes 16 bits of node number in
 order to avoid these problems.  We plan to remove this requirement in the
-future to enable --llvm-wide-opt to work with CHPL_WIDE_POINTERS=struct.
+future to enable ``--llvm-wide-opt`` to work with CHPL_WIDE_POINTERS=struct.
 
 Caveats:
-* --llvm-wide-opt currently requires CHPL_WIDE_POINTERS=node16, which means
+
+* ``--llvm-wide-opt`` currently requires CHPL_WIDE_POINTERS=node16, which means
```
Comment (ben-albrecht) [inline doc/release/technotes/README.llvm.rst]: \`\`CHPL_WIDE_POINTERS=node16\`\`

File: doc/release/technotes/README.llvm.rst (hunk 1 above)
Comment (ben-albrecht) [inline doc/release/technotes/README.llvm.rst]: Couple of other spots too. I'll let you get your next commit in, and see if I spot anymore after that.

**Comment by ben-albrecht** [comment]
Looks good other than some spots that need \`\`s

File: doc/release/technotes/README.llvm.rst (hunk 2 above)
Comment (mppf) [inline doc/release/technotes/README.llvm.rst]: I don't know what you mean. Do you want clang to be in `` ? the --ccflags semes OK...

File: doc/release/technotes/README.llvm.rst (hunk 1 above)
Comment (mppf) [inline doc/release/technotes/README.llvm.rst]: Yeah, have a look now.

**Comment by mppf** [comment]
@ben-albrecht - could you have a quick 2nd look? I think it's good now...

**Comment by ben-albrecht** [comment]
@mppf Yep - looks great now! :+1:

**Comment by ronawho** [comment]
@mppf do you mind adding

```
.. _readme-llvm:

```

to the top of this file before merging?

**Pull request merged**